h2o-genmodel/src/main/java/hex/genmodel/tools/BuildPipeline.java

```diff
@@ -0,0 +1,157 @@
+    for (String alias : _input.keySet()) {
+      if (! subModels.contains(alias)) {
+        candidates.add(alias);
+      }
+    }
+    if (candidates.size() != 1) {
+      throw new IllegalStateException("Main model cannot be identified, " +
+              "main should be the only model that doesn't have output mappings. Candidates: " + candidates.toString());
+    }
+    return candidates.iterator().next();
+  }
+
+  private static void usage() {
+    System.out.println("");
+    System.out.println("Usage:  java [...java args...] hex.genmodel.tools.BuildPipeline ");
+    System.out.println("             --mapping <inputMapping1> <inputMapping2> ... --output <outputFile> --input <inputFile1> <inputFile2> ...");
+    System.out.println("");
+    System.out.println("     --mapping Mapping of model predictions to main model inputs.");
+    System.out.println("               Example: to map the predicted cluster (output 0) of a model defined in MOJO 'clustering.zip'");
+    System.out.println("                        to input column 'CLUSTER' of the main model, use specification 'CLUSTER=cluster:0'.");
+    System.out.println("     --input   List of input MOJO files representing both the main model and the prerequisite models.");
+    System.out.println("     --output  Name of the generated MOJO pipeline file.");
+    System.out.println("");
+    System.out.println("     Input mappings are specified in format '<columnName>=<modelAlias>:<predictionIndex>'.");
+    System.out.println("     Model alias is based on the name of the MOJO file, MOJO stored in 'glm_model.zip' will have alias 'glm_model'.");
+    System.out.println("");
+    System.out.println("Note: There is no need to specify which of the MOJO model represents the main model. The tool");
+    System.out.println("automatically identifies the main model as the one that doesn't have any output mappings.");
+    System.out.println("");
+    System.exit(1);
+  }
+
+  private void parseArgs(String[] args) {
+    try {
+      for (int i = 0; i < args.length; i++) {
+        String s = args[i];
+        if (s.equals("--mapping")) {
+          List<String> mappingSpec = readArgValues(args, i + 1);
+          _mappings = new ArrayList<>(mappingSpec.size());
+          for (String spec : mappingSpec) {
+            try {
+              _mappings.add(MojoPipelineBuilder.MappingSpec.parse(spec));
+            } catch (Exception e) {
+              throw new IllegalArgumentException("Invalid mapping specified ('" + spec + "'." +
+                      " Please use format '<columnName>=<modelAlias>:<predictionIndex>'.");
+            }
+          }
+          i += mappingSpec.size();
+        } else if (s.equals("--output")) {
+          List<String> outputFile = readArgValues(args, i + 1);
+          if (outputFile.size() != 1) {
+            throw new IllegalArgumentException("Invalid specification of the output file. Please specify a single output file.");
```

Comment:
For debugging purpose, it makes our life easier to print what was provided by the user. In this case the length of outputFile and maybe the content as well. Might make sense in other cases as well.

Reply:
good suggestion, will add